Sources/ComposableArchitecture/SharedState/PersistenceKey/AppStorageKey.swift

```diff
@@ -300,16 +300,23 @@ extension AppStorageKey: PersistenceKey {
       else { return }
       didSet(self.store.value(forKey: self.key) as? Value ?? initialValue)
     }
-    let willEnterForeground = NotificationCenter.default.addObserver(
-      forName: willEnterForegroundNotificationName,
-      object: nil,
-      queue: nil
-    ) { _ in
-      didSet(self.store.value(forKey: self.key) as? Value ?? initialValue)
+    let willEnterForeground: (any NSObjectProtocol)?
+    if let willEnterForegroundNotificationName {
```

Comment:
Since these notification names are `nil` for platforms that don't support the notification we should unwrap them. Otherwise we subscribe to notifications with a `nil` name, which basically means subscribe to _all_ notifications.